test_util/launch.py

```diff
@@ -1,20 +1,288 @@
+                dcos_dns=test_host,
+                master_list=[m['private_ip'] for m in details['masters']],
+                agent_list=[a['private_ip'] for a in details['private_agents']],
+                public_agent_list=[a['private_ip'] for a in details['public_agents']],
+                aws_access_key_id=info['provider']['access_key_id'],
+                aws_secret_access_key=info['provider']['secret_access_key'],
+                region=info['provider']['region'],
+                test_cmd=test_cmd)
+
+    def get_stack(self, info):
+        """Returns the correct class interface depending how the AWS CF is configured
+        NOTE: only supports Simple Cloudformation currently
+        """
+        try:
+            return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+        except Exception as ex:
+            raise LauncherError('StackNotFound', '{} is not accessible'.format(info['stack_name'])) from ex
+
+
+def get_launcher(launcher_type, provider_info):
+    """Returns a launcher given a launcher type (string) and a dictionary of
+    appropriate provier_info
+    """
+    if launcher_type == 'cloudformation':
+        check_keys(provider_info, ['region', 'access_key_id', 'secret_access_key'])
+        boto_wrapper = BotoWrapper(
+            provider_info['region'], provider_info['access_key_id'], provider_info['secret_access_key'])
+        return AwsCloudformationLauncher(boto_wrapper)
+    else:
+        raise LauncherError('UnsupportedAction', 'Launcher type not supported: {}'.format(launcher_type))
+
+
+def convert_host_list(host_list):
+    """ Makes Host tuples more readable when using describe
+    """
+    return [{'private_ip': h.private_ip, 'public_ip': h.public_ip} for h in host_list]
+
+
+def check_keys(user_dict, key_list):
+    missing = [k for k in key_list if k not in user_dict]
+    if len(missing) > 0:
+        raise LauncherError('MissingInput', 'The following keys were required but '
+                            'not provided: {}'.format(repr(missing)))
+
+
+def do_main(args):
+    if args['create']:
+        info_path = args['--info-path']
+        if os.path.exists(info_path):
+            raise LauncherError('InputConflict', 'Target info path already exists!')
+        config = load_yaml(args['--config-path'])
+        check_keys(config, ['type', 'provider_info', 'this_is_a_temporary_config_format_do_not_put_in_production'])
+        write_json(info_path, get_launcher(config['type'], config['provider_info']).create(config))
+        return 0
+
+    info = load_json(args['--info-path'])
+    check_keys(info, ['type', 'provider'])
+    launcher = get_launcher(info['type'], info['provider'])
+
+    if args['wait']:
+        launcher.wait(info)
+        print('Cluster is ready!')
+        return 0
+
+    if args['describe']:
+        print(json_prettyprint(launcher.describe(info)))
+        return 0
+
+    if args['pytest']:
+        test_cmd = 'py.test '
+        if args['--env'] is not None:
```

Comment:
Let's `assert '=' not in args['--env']` here so that the command fails ASAP if someone mistakenly tries to define an env var within the `dcos-launch pytest` invocation.